provider/pkg/provider/handlers.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2021, Pulumi Corporation.
+
+package provider
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/imdario/mergo"
+	"github.com/pkg/errors"
+	"github.com/pulumi/pulumi-google-native/provider/pkg/resources"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/logging"
+	rpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
+)
+
+type updateHandler interface {
+	Update(inputs resource.PropertyMap, oldState resource.PropertyMap) (map[string]interface{}, error)
+}
+
+type updateHandlerFunc func(inputs resource.PropertyMap, oldState resource.PropertyMap) (map[string]interface{}, error)
+
+func (u updateHandlerFunc) Update(inputs resource.PropertyMap, oldState resource.PropertyMap) (map[string]interface{}, error) {
+	return u(inputs, oldState)
+}
+
+func (p *googleCloudProvider) updateHandler(resourceTok string, req *rpc.UpdateRequest) updateHandler {
+	logging.V(9).Infof("UpdateHandler for %q", resourceTok)
+	switch resourceTok {
+	case "google-native:container/v1:Cluster", "google-native:container/v1beta1:Cluster":
+		return updateHandlerFunc(func(inputs, oldState resource.PropertyMap) (map[string]interface{}, error) {
+			res, ok := p.resourceMap.Resources[resourceTok]
+			if !ok {
+				return nil, errors.Errorf("resource %q not found", resourceTok)
+			}
+
+			updateBody := p.prepareAPIInputs(inputs, oldState, res.UpdateProperties)
+
+			// TODO: Cluster updates also include flattened members of the `networkConfig` map
```

Comment:
See https://github.com/pulumi/pulumi-google-native/blob/master/discovery/container_v1.json#L3029 for examples of fields that are originally specified under `networkConfig` in create properties but are flattened in the update request.

In addition, the update also is not sufficient to adequately cover all mutable fields. See `updateMaster` for an example: https://github.com/pulumi/pulumi-google-native/blob/master/discovery/container_v1.json#L3029